docs/source/notes/autograd.rst

```diff
@@ -210,3 +210,102 @@ No thread safety on C++ hooks
 Autograd relies on the user to write thread safe C++ hooks. If you want the hook
 to be correctly applied in multithreading environment, you will need to write
 proper thread locking code to ensure the hooks are thread safe.
+
+Autograd for Complex Numbers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+What notion of complex derivative does PyTorch use?
+***************************************************
+
+PyTorch follows `JAX's <https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#Complex-numbers-and-differentiation>`_
+convention for autograd for Complex Numbers.
+
+Suppose we have a function :math:`F: C -> C` which we can decompose into functions u and v
+which compute the real and imaginary parts of the function:
+
+    .. code::
+
+        def F(z):
+            x, y = real(z), imag(z)
+            return u(x, y) + v(x, y) * 1j
+
+The Jacobian, JVP and VJP for function :math:`F` at :math:`(x, y)` are defined as:
+
+    .. code::
+
+        def JVP(tangent):
+            c, d = real(tangent), imag(tangent)
+            return [1, i]^T * J * [c, d]
+
+    .. code::
+
+        def VJP(cotangent):
+            c, d = real(cotangent), imag(cotangent)
+            return [c, -d]^T * J * [1, -i]
+
+    where
+
+    .. math::
+
+        J = \begin{bmatrix}
+            \partial_0u(x, y) & \partial_1u(x, y)\\
+            \partial_0v(x, y) & \partial_1v(x, y) \end{bmatrix} \\
+
+In PyTorch, the VJP is mostly what we care about, as it is the computation performed when we do backward
+mode automatic differentiation. Notice that d and i (imaginary number) are negated in the formula above.
+
+Why is there a negative sign in the formula above?
+**************************************************
+
+For a function F: V → W, where are V and W are vector spaces. The output of
+the Vector-Jacobian Product :math:`VJP : V → (W^* → V^*)` is a linear map
+from :math:`W^* → V^*` (explained in `Chapter 4 of Dougal’s thesis <https://dougalmaclaurin.com/phd-thesis.pdf>`_).
+
+The negative signs in the above VJP computation are due to conjugation. The first
+vector in the output returned by VJP for a given cotangent is a covector (\in :math:`C^*`),
+and the last vector in the output is used to get the result in :math:`C`
+since the final result of reverse-mode differentiation of a function is a covector belonging
+to :math:`C^*` (explained in `Chapter 4 of Dougal’s thesis <https://dougalmaclaurin.com/phd-thesis.pdf>`_).
+
+What happens if I call backward() on a complex scalar?
+******************************************************
+
+For geneneral ℂ→ℂ functions, the Jacobian has 4 real-valued degrees of freedom (as in the 2x2 Jacobian matrices above),
```

Comment:
I think this is missing one extra step:
The backward in pytorch does not compute a jacobian.

I think you want something that mentions that for R->R and C->R, backward() (with no argument) computes the full gradient of the function.
But for C->C functions, this is not case...

Reply:
Yeah agreed but we are not saying the backward is computing the grad. We are just commenting on the Jacobian that's used to summarize that why the gradient for holomorphic functions can still be represented a s a complex number

Reply:
Ok, then I guess the title of the section is what confused me here.